src/browser/api/window.js

```diff
@@ -1051,25 +1051,95 @@ Window.getParentApplication = function() {
 Window.getParentWindow = function() {};
 
 /**
- * Fetches window's preload script and gets its content
+ * Fetches window's preload script(s) to cache and reads in from cache.
+ * ```typescript
+ * type Preload = string | Preloader;
+ * interface Preloader { url:string; optional?:boolean; name?:string; };
+ * interface PreloadPayload extends Preloader { script:string, index:number, description:string };
+ * ```
+ * @param identity {{uuid:string}}
+ * @param preloads {Preload|Preload[]}
+ * @param ack {function(PreloadPayload[])} - Called once with all script successfully fetched & read.
+ * @param nack {function(Error)} - Called on first required fetch or read failed.
  */
-Window.getPreloadScript = function(identity, preloadUrl, callback) {
-    cachedFetch(identity.uuid, preloadUrl, (fetchError, scriptPath) => {
-        if (fetchError) {
-            return callback(new Error(`Failed to fetch preload script from ${preloadUrl}`));
+Window.getPreloadScript = function(identity, preloads, ack, nack) {
+    if (!preloads || !preloads.length) {
+        return ack([]); //shortstop
```

Comment:
overkill but if caller really wants that they can give something like this for the `ack` param: `(arr) => !arr.length`